code/obj/item/device/lights.dm

```diff
@@ -735,3 +735,155 @@ TYPEINFO(/obj/item/device/light/floodlight)
 	rotatable = FALSE
 	infinite_power = TRUE
 	power_usage = 0 WATTS
+
+#define FLARE_UNLIT 1
+#define FLARE_LIT 2
+#define FLARE_BURNT 3
+
+/obj/item/roadflare
+	name = "emergency flare"
+	desc = "Space grade emergency flare that can burn in an 02 free environment. Estimated burn time 3-6 minutes."
+	icon = 'icons/obj/lighting.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "roadflare"
+	uses_multiple_icon_states = 1
+	w_class = W_CLASS_POCKET_SIZED
+	throwforce = 1
+	flags = FPRINT | TABLEPASS
+	stamina_damage = 0
+	stamina_cost = 0
+	stamina_crit_chance = 1
+	burn_point = 220
+	burn_output = 1200
+	burn_possible = 1
+
+	var/on = FLARE_UNLIT
+
+	var/life_time = 0
+	rand_pos = 1
+
+	var/col_r = 0.95
+	var/col_g = 0.7
+	var/col_b = 0.25
+	var/brightness = 0.4
+	var/datum/component/loctargeting/medium_light/light_c
+
+	New()
+		..()
+		light_c = src.AddComponent(/datum/component/loctargeting/medium_light, col_r*255, col_g*255, col_b*255, 510 * brightness)
+		light_c.update(0)
+
+
+	process()
+		if (src.on == FLARE_LIT)
+			if (world.time > life_time)
+				var/location = src.loc
+				if (ismob(location))
+					var/mob/M = location
+					src.put_out(M)
+					return
+				else
+					src.put_out()
+					return
+			var/turf/T = get_turf(src.loc)
+			if (T)
+				T.hotspot_expose(900,5)
+
+	proc/light(var/mob/user as mob)
+		src.on = FLARE_LIT
+		w_class = W_CLASS_BULKY
+		src.firesource = FIRESOURCE_OPEN_FLAME
+		src.icon_state = "roadflare-lit"
+
+		playsound(user, 'sound/items/matchstick_light.ogg', 80, FALSE)
+		light_c.enable()
+
+		src.life_time = (world.time + rand(1800,3600))
+		processing_items |= src
+		if (istype(user))
+			user.update_inhands()
+		src.UpdateParticles(new/particles/roadflare_smoke,"roadflare_smoke")
+
+	proc/put_out(var/mob/user as mob)
+		src.on = FLARE_BURNT
+		w_class = W_CLASS_POCKET_SIZED
+		src.firesource = FALSE
+		src.icon_state = "roadflare-burnt"
+		src.item_state = "roadflare"
+		src.name = "burnt-out emergency flare"
+
+		playsound(src, 'sound/impact_sounds/burn_sizzle.ogg', 70, FALSE)
+		light_c.disable()
+		if (istype(user))
+			user.update_inhands()
+		processing_items.Remove(src)
+		src.ClearSpecificParticles("roadflare_smoke")
+
+	temperature_expose(datum/gas_mixture/air, temperature, volume)
+		if (src.on == FLARE_UNLIT)
+			if (temperature > T0C+200)
+				src.visible_message("<span class='alert'>[src] ignites!</span>")
+				src.light()
+
+	ex_act(severity)
```

Comment:
This should call the parent, otherwise these will be indestructible to explosions (lighting off small explosions makes sense but it seems odd that a nuke wouldn't destroy them lol)

Reply:
Forgot plus this is based on a match so lol, matches are nuke proof